src/components/Chat/BottomSheetGif.js

```diff
@@ -0,0 +1,176 @@
+import * as React from 'react';
+import {
+  View,
+  Text,
+  StyleSheet,
+  Dimensions,
+  TouchableOpacity,
+  TextInput,
+  FlatList
+} from 'react-native';
+import FastImage from 'react-native-fast-image';
+
+import {fonts, normalizeFontSize} from '../../utils/fonts';
+import {colors} from '../../utils/colors';
+import {BottomSheet} from '../BottomSheet';
+import MemoIcSearch from '../../assets/icons/Ic_search';
+import dimen from '../../utils/dimen';
+import {SIZES} from '../../utils/theme';
+import IconClear from '../../assets/icon/IconClear';
+import {getGifFeatured} from '../../service/gif';
+
+const {height: heightScreen, width: widthScreen} = Dimensions.get('window');
+
+const BottomSheetGif = React.forwardRef((props, ref) => {
+  const {onCancel, onSelect} = props;
+  const [searchText, setSearchText] = React.useState('');
+  const [list, setList] = React.useState([]);
+  const [isLoading, setIsLoading] = React.useState(true);
+
+  const handleChangeText = (text) => {
+    setSearchText(text);
+  };
+
+  const handleOnClearText = () => {
+    setSearchText('');
+  };
+
+  const onLoad = async () => {
+    setIsLoading(true);
+    try {
+      const result = await getGifFeatured();
+      if (result.code === 200) {
+        setList(result.data);
+      }
+    } catch (error) {
+      console.error(error);
+    }
+    setIsLoading(false);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `onLoad` function is fetching GIFs from an external service. However, there's no error handling for network issues or unexpected responses. Consider adding a more robust error handling mechanism to improve the user experience in case of failures.
```diff
  const onLoad = async () => {
    setIsLoading(true);
    try {
      const result = await getGifFeatured();
      if (result.code === 200) {
        setList(result.data);
      } else {
        // Handle non-200 response codes
        console.error(`Error: Received status code ${result.code}`);
      }
    } catch (error) {
      console.error(error);
      // Handle network errors
      console.error('Network error while fetching GIFs');
    }
    setIsLoading(false);
  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->